xbmc/music/MusicThumbLoader.cpp

```diff
@@ -322,6 +322,17 @@ bool CMusicThumbLoader::FillLibraryArt(CFileItem &item)
             bDiscSetThumbSet = true;
           }
         }
+        else if (StringUtils::StartsWith(artitem.artType, "discart"))
+        {
+          int number = atoi(artitem.artType.substr(7).c_str());
+          if (number == 0)
+            item.SetArtFallback("discart", artname);
+          else if (number > 0 && tag.GetDiscNumber() == number)
```

Comment:
Hard coded art type again. What is needed muct work regardless of what the disc art gets called by scraper, skin or user, and even support multiple types of disc specific art (OK I can only think of a CD image, but who knows what furture things we could have??)

Reply:
Take a look at my reworking, we possibly cross posted.

Reply:
Yeah, I will.  In regard to the hard-coding, I can see no other way with what there is at the moment to set a specific 'something%i', where %i may or not exist, into Player.Art(discart), which is what a skin is expecting.  In the art table, it is of type 'Album' and AS.xml has artistextraart and albumextraart.  If however, there was a type of 'Disc' and discextraart, then one could distinguish discart from anything else by the type.  Otherwise, I don't see how to separate out discart from thumb or cover or any other string that could have a numerical suffix and is of type Album without having a hard-coded string in there.  Thats before I've looked at what you've done though, so perhaps I'm talking without my book !!  

I didn't think that you would go for this anyway :)  

Just out of interest, is there any way of deleting art from the art table when something changes, without editing the db ?  In testing I ended up with discart and discart1 for a couple of albums and found that discart was always displayed (as it is found first) and the only way to change that was edit the db and drop that row......or at least, I think it was the only way.

Reply:
Ha, well looks like I was completely wrong then.  Just tested with 4 different skins with albums with multiple and single discs and all displayed correctly.  Now I need to go and pick your code apart to see how you've done it.

Reply:
Yes, pick that code apart, see if you can make sense of it or find more design omissions or mistakes :)

Try to think of art generically. When I say disc art I mean art associated with subpart of a release, an obvious example is the round CD image which was "cdart" for some addons/skins but we are suggesting standardising on "discart". However parts can also have inner sleave covers (taken from folder.jpg and held as "thumb") and could have other stuff in the future - wild example could be sleave back. Core is stuck with thumb (a defualt image of some kind), but does not want to hard code anything more.

Current skins may look for "discart", and scraper may fetch "discart" and local files called discart.png  will be picked up as "discart" if the right seeing is added to as.xml. The same core processing needs to work for "foobar" if that is what users/skins/scraper choose to implement as an art type.

Further we are treating a disc like a subset of an album with all the same properties plus a number and subtitle, hence same goes for art. Any art types an album can have ("thumb" + whatever is configured in as.xml) a disc can have, and are held as if for the album with disc number appended e.g. "thumb2", "foobar3" etc. So if a type of album art has a numberic suffix then it is disc (album part)  specific. What you reminded me was that the song needed its specific part of album art too.

> is there any way of deleting art from the art table when something changes,

From album info dialog, select art, or rather deselect that type. Of course all the possible art types (based on what types you have for some albums) get shown even when that specific item does not have that art as a way to allow users to select manually an use art type consistently. It is all still rather raw as a UI.